### DISCUSSION TITLE
How to add files that are not Python

### DISCUSSION THREAD

**Discussion by FlxVctr** [beeware/beeware#119] [Q&A]
Hi,
I am struggling to learn from the docs how to include a file in the app (e.g. a config or data file that the app has to read in order to function). I tried to add a file to the src folder (in an app that was created following the beeware tutorial), which I can use via the 'src/appname/src/file' path in dev mode. However, as soon as I build the application I get an file not found error.

Also tried the resources folder, but no luck. Can someone give me a hint how to package and reference a file for the app correctly?

**Comment by freakboy3742** [comment]
The Briefcase configuration has a `sources` section; that lists folders (and individual files) that will be copied into the final app. That copy process is recursive; so as long as your config/data file is contained within the hierarchy of the folder listed in `sources`, it will be copied into the app. You can check this by looking for the `app` folder in the generated platform-specific project.

The issue then is *locating* the file. The "working directory" for the app is different on almost every platform, so relative path locations  will be unreliable; you need to use an absolute path based on somewhere that is known. The easiest location is usually the `__file__` attribute of the app code itself; you can then build a path from the known location of the config/data file.

If you're using Toga, you can get this path pre-computed for you as the `App.paths.app` attribute (i.e., your app has a `paths` object that contains a bunch of useful paths for your app to use for storing files, logs, etc.

**Reply by FlxVctr** [reply]
Thanks! The App.paths attribute works for me. Couldn't find that in the toga documentation though. Maybe worth including?

**Reply by FlxVctr** [reply]
Also, would like to mark as answer but Github/my browser seems to exhibit a bug that doesn't let me -_-
